### PR TITLE
adding learndl template and localization

### PR DESCRIPTION
**Pull request by lloydwaddell**
**merges into learndl/main branch**

### Context
Drafting practice template for a new practice type `Learndl`. 

Based template schema off of mockup : https://d2l.slack.com/archives/C03H5VDGHAP/p1654541374367169
Schema contains:
- title
- description
- instructions
- new type `learndl`
- number of guesses
- feedback

Data structure contains 3 lists: answers, attributes (column names), attribute values which contains relationship of answer -> attribute

### CODE REVIEW

File: components/practice/d2l-practices.js [hunk 1]
```diff
@@ -261,6 +261,154 @@ const practiceTemplates = function(localize) {
...
+				'scorePoints': 1,
+				'scoreCriteria': 0,
+				'scoreCalculation': 0,
+				'attributes':[
+					{ 'id': 0, 'title': localize('practice:learndl:attributes', { index: 1 }), 'type': 0, 'image': '', 'alt': '' },
+					{ 'id': 1, 'title': localize('practice:learndl:attributes', { index: 2 }), 'type': 0, 'image': '', 'alt': '' },
+					{ 'id': 2, 'title': localize('practice:learndl:attributes', { index: 3 }), 'type': 0, 'image': '', 'alt': '' },
+					{ 'id': 3, 'title': localize('practice:learndl:attributes', { index: 4 }), 'type': 0, 'image': '', 'alt': '' }
+				],
+				'learndlAnswers':[
```
Comment (louiscoste) [inline components/practice/d2l-practices.js:289]: nit:
```suggestion
				'answers':[
```

File: components/practice/d2l-practices.js [hunk 2]
```diff
@@ -261,6 +261,154 @@ const practiceTemplates = function(localize) {
...
+					{ 'id':0, 'title': localize('practice:learndl:answers'), 'image':'', 'alt': '', 'correctFeedback': '', 'incorrectFeedback': '' },
```
[the remark this answers is not in the document]
Comment (louiscoste) [reply components/practice/d2l-practices.js:290]: I am not sure images for the answers makes sense...

File: components/practice/d2l-practices.js [hunk 3]
```diff
@@ -261,6 +261,154 @@ const practiceTemplates = function(localize) {
...
+							type: 'learndl',
+							label: '',
+							required: false,
+							value: 0
+						}
+					],
+					key: 0
+				},
+				{
+					class: 'd2l-guesses',
```
Comment (louiscoste) [inline components/practice/d2l-practices.js:376]: nit: I realize I did put the number of guesses below the table, but I meant to have it above the table (I just forgot to add before drawing the table...)

File: components/practice/d2l-practices.js [hunk 4]
```diff
@@ -261,6 +261,154 @@ const practiceTemplates = function(localize) {
...
+			tag: '',
+			value: 2,
+			default: {
+				'id': '',
+				'type': 2,
+				'title': '',
+				'description': '',
+				'customInstructions': '',
+				'displayFeedback': false,
+				'showCategoryCountdown': false,
```
[the remark this answers is not in the document]
Comment (lloydwaddell) [reply components/practice/d2l-practices.js:276]: I am not sure, but the other templates have it as well. I plan to do a hardening of properties once the component is in place and have a better understanding of what is needed.

**Review by louiscoste — APPROVED** [review]

**Comment by grant-cleary** [comment]
You'll need to sync lang terms for linting to pass. `npm run langs:sync` should do the job for you.

**Comment by lloydwaddell** [comment]
> You'll need to sync lang terms for linting to pass. `npm run langs:sync` should do the job for you.

oh nice, I started to do that manually, but I will restore and run this command

**Comment by grant-cleary** [comment]
> > You'll need to sync lang terms for linting to pass. `npm run langs:sync` should do the job for you.
> 
> oh nice, I started to do that manually, but I will restore and run this command

Haha yeah, it's a huge pain otherwise. Too much copying/pasting and inevitably there will be a typo somewhere that still fails linting...

**Review by grant-cleary — APPROVED** [review]

**Pull request merged**